internal/worker/server.go

```diff
@@ -40,6 +40,9 @@ const (
 	JobTypeAWSEC2Share      string = "aws-ec2-share"
 )
 
+const maxHeartbeatRetries = 2
```

Comment:
Why don't just use the same retry count for both failure cases? This seems a bit counter-intuitive with regard to the current implementation. E.g. After 2 interrupt retries, no heartbeat retry will actually happen as we have only one retry counter.

Reply:
Fair, let's not try to be clever. If we need this to be more fine-grained we can update that in the future. For now I upped the retry count a bit and used the same in both cases.